From f1d1b55d82e9f2d77c5b54422bced31f44370ce0 Mon Sep 17 00:00:00 2001
From: Ben <bnj@ewft.org>
Date: Sun, 19 Jan 2020 18:28:16 +0100
Subject: [PATCH] Fix PHP-7.4 Issue

PHP-7.4 Issue  = "Trying to access array offset on value of type null" in functions_category.inc.php
---
 include/functions_category.inc.php | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/include/functions_category.inc.php b/include/functions_category.inc.php
index 22997ee2f..64925a0c8 100644
--- a/include/functions_category.inc.php
+++ b/include/functions_category.inc.php
@@ -103,6 +103,8 @@ function get_categories_menu()
   $result = pwg_query($query);
   $cats = array();
   $selected_category = isset($page['category']) ? $page['category'] : null;
+  $selected_category_id = isset($selected_category['id']) ? $selected_category['id'] : null;
+  $selected_category_id_uppercat = isset($selected_category['id_uppercat']) ? $selected_category['id_uppercat'] : null;
   while ($row = pwg_db_fetch_assoc($result))
   {
     $child_date_last = @$row['max_date_last']> @$row['date_last'];
@@ -122,8 +124,8 @@ function get_categories_menu()
           ),
         'URL' => make_index_url(array('category' => $row)),
         'LEVEL' => substr_count($row['global_rank'], '.') + 1,
-        'SELECTED' => $selected_category['id'] == $row['id'] ? true : false,
-        'IS_UPPERCAT' => $selected_category['id_uppercat'] == $row['id'] ? true : false,
+        'SELECTED' => $selected_category_id == $row['id'] ? true : false,
+        'IS_UPPERCAT' => $selected_category_id_uppercat == $row['id'] ? true : false,
         )
       );
     if ($conf['index_new_icon'])
@@ -602,4 +604,4 @@ function remove_computed_category(&$cats, $cat)
   unset($cats[$cat['cat_id']]);
 }
 
-?>
\ No newline at end of file
+?>
